src/librustc_trans/adt.rs

```diff
@@ -900,8 +684,7 @@ pub fn trans_set_discr<'blk, 'tcx>(bcx: Block<'blk, 'tcx>, r: &Repr<'tcx>,
 fn assert_discr_in_range(ity: IntType, min: Disr, max: Disr, discr: Disr) {
     match ity {
         attr::UnsignedInt(_) => {
-            assert!(min <= discr);
-            assert!(discr <= max);
+if min <= max { assert!(min <= discr && discr <= max) } else { assert!(min <= discr || discr <= max) }
```

Comment:
This function shouldn't take an `IntType` and the `if`/`else` should be on multiple lines and properly indented.